File: project-euler-100-java/src/main/java/com/github/adrianomoreira/projecteuler100/Problem009.java
package com.github.adrianomoreira.projecteuler100;

public class Problem009 {
    public static int specialPythagoreanTriplet(int n) {
        for (int c = n - 2; c > 0; c--) {
            for (int b = c - 1; b > 0; b--) {
                for (int a = b - 1; a > 0; a--) {
                    if (a + b + c == n) {
                        if (isPythagorean(a, b, c)) {
                            return a * b * c;
                        }
                    }
                }
            }
        }

        throw new RuntimeException("not found");
    }

    private static boolean isPythagorean(int a, int b, int c) {
        return Math.pow(a, 2) + Math.pow(b, 2) == Math.pow(c, 2);
    }
}

File: project-euler-100-go/problem_009.go
package main

import (
	"math"
)

func isPythagorean(a int, b int, c int) bool {
	return math.Pow(float64(a), 2)+math.Pow(float64(b), 2) == math.Pow(float64(c), 2)
}

func specialPythagoreanTriplet(n int) int {
	for c := n - 2; c > 0; c-- {
		for b := c - 1; b > 0; b-- {
			for a := b - 1; a > 0; a-- {
				if a+b+c == n {
					if isPythagorean(a, b, c) {
						return a * b * c
					}
				}
			}
		}
	}
	return 0
}
